File: LeetCode/Arrays/1 - Two Sum/Solution.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        if not nums or len(nums) == 0:
            return []

        hashMap = {}

        for index, num in enumerate(nums):
            if(num in hashMap):
                return [index, hashMap[num]]
            else:
                hashMap[target - num] = index
        return

File: LeetCode/Arrays/1 - Two Sum/Solution.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
  if (!nums || nums.length === 0) {
    return [];
  }

  const hashMap = new Map();
  for (let i = 0; i < nums.length; i++) {
    if (hashMap.has(nums[i])) {
      return [i, hashMap.get(nums[i])];
    }
    hashMap.set(target - nums[i], i);
  }
  return [];
}